fpa/grpc/service.go

```diff
@@ -104,9 +112,24 @@ func (s *Service) AddCid(ctx context.Context, req *pb.AddCidRequest) (*pb.AddCid
 	if err != nil {
 		return nil, err
 	}
-	if err := i.AddCid(ctx, c); err != nil {
+	log.Infof("adding cid %s", c)
+	jid, err := i.AddCid(c)
+	if err != nil {
 		return nil, err
 	}
+
+	ch, err := i.Watch(jid)
+	if err != nil {
+		return nil, fmt.Errorf("watching add cid created job: %s", err)
+	}
+	defer i.Unwatch(ch)
+	for job := range ch {
+		if job.Status == fpa.Done {
+			break
+		} else if job.Status == fpa.Cancelled || job.Status == fpa.Failed {
+			return nil, fmt.Errorf("error adding cid: %s", job.ErrCause)
+		}
+	}
```

Comment:
Since now pinning a Cid is an async process and to avoid that change going down the path of the CLI yet, the service watches the returned _Job_ until:
- fails or gets cancelled (a state that yet doesn't happen, but maybe will): in which case, returns the error cause.
- succeeds: all good.

So the grpc api still behaves async. I'll wait a bit before making changes to the CLI until the next PR that allows setting configs on the pinnings and retrieval gets in shape, to avoid wasted work and think about its final shape too.